### PR TITLE
Removes verifyStorageConfiguration from tools/lxdclient and implement…

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This moves the check for a disk device in the default LXD profile from to-be-deprecated tools/lxdclient and implements in container/lxd.

## QA steps

- New unit tests.
- Delete the default NIC and disk devices from the profile, bootstrap and ensure that these are re-created.
- Bootstrap again to ensure that behaviour is correct when they are present.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
[Remove Old LXD Client and Use New LXD Server Directly in LXD Provider](https://trello.com/c/uKWwlPMh/314-remove-old-lxd-client-and-use-new-lxd-server-directly-in-lxd-provider)

File: container/lxd/storage_test.go [hunk 1]
```diff
@@ -86,3 +101,55 @@ func (s *storageSuite) TestCreateVolume(c *gc.C) {
...
+		cSvr.EXPECT().CreateStoragePool(req).Return(nil),
+		cSvr.EXPECT().UpdateProfile("default", profile.Writable(), lxdtesting.ETag).Return(nil),
+	)
+
+	jujuSvr, err := lxd.NewServer(cSvr)
+	c.Assert(err, jc.ErrorIsNil)
+
+	profile.Devices = nil
+	c.Assert(jujuSvr.EnsureDefaultStorage(profile, lxdtesting.ETag), jc.ErrorIsNil)
+}
```
Comment (SimonRichardson) [inline container/lxd/storage_test.go]: We should add tests around the logic of if the poolName is empty...

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
This one again.
```
Adding juju PPA for mongodb
gpg: keyring `/tmp/tmpmfudqmko/secring.gpg' created
gpg: keyring `/tmp/tmpmfudqmko/pubring.gpg' created
gpg: requesting key C8068B11 from hkp server keyserver.ubuntu.com
gpgkeys: key A2EB2DEC0BD7519B7B38BE38376A290EC8068B11 can't be retrieved
gpg: no valid OpenPGP data found.
gpg: Total number processed: 0
gpg: keyserver communications error: keyserver helper general error
gpg: keyserver communications error: unknown pubkey algorithm
gpg: keyserver receive failed: unknown pubkey algorithm
Failed to add key.
make: *** [install-dependencies] Error 1
Makefile:115: recipe for target 'install-dependencies' failed
```

$$merge$$
